rust/trace/src/lib.rs

```diff
@@ -819,6 +821,22 @@ impl Trace {
         samples.sort_unstable();
         samples
     }
+
+    pub fn save<P: AsRef<Path>>(
```

Comment:
I'd make this `pub(crate)` since we don't expect it to be used externally. One important reaeson to avoid `pub` when not necessary is that it makes it a semvar breaking change if we want to modify the api in the future.